### ISSUE TITLE
Repeats one SVG twice instead of including second SVG

### ISSUE THREAD

**Issue by donut** [jkphl/svg-sprite#96]
I have two SVGs that are the same shape and size but different colors. When I add them to the sprite sheet they both show up as the lighter variation instead of one dark and one light.

[next-arrow.svg](https://dl.dropboxusercontent.com/u/416667/next-arrow.svg) is missing.
![next-arrow.svg](https://dl.dropboxusercontent.com/u/416667/next-arrow.svg)

[next-arrow-hover.svg](https://dl.dropboxusercontent.com/u/416667/next-arrow-hover.svg) is appearing twice.
![next-arrow-hover.svg](https://dl.dropboxusercontent.com/u/416667/next-arrow-hover.svg)

 [sprites-4c00f527.svg](https://dl.dropboxusercontent.com/u/416667/sprites-4c00f527.svg) is what gulp-svg-sprite generated.
![sprites-4c00f527.svg](https://dl.dropboxusercontent.com/u/416667/sprites-4c00f527.svg)

[zip of files](https://dl.dropboxusercontent.com/u/416667/Problem%20SVGs.zip)

This wasn't happening when they weren't exactly the same size.

SVGs exported from Adobe Illustrator CC 2015. gulp-svg-sprite 1.2.4

Settings being passed to gulp-svg-sprite:

```js
svg_sprite({
  shape: {
    spacing: {
      padding: 35
    }
  },
  mode: {
    css: {
      dest: '',
      sprite: "sprites.svg",
      render: {
        styl: {
          template: 'svg_sprites_template.styl',
          dest: "sprites-svg.styl"
        }
      }
    }
  }
});
```

Output from verbose logging:
```
2015-05-17 10:55:30 - verbose: Initialized spriter configuration
2015-05-17 10:55:30 - info: Created spriter instance
2015-05-17 10:55:30 - verbose: Added shape "slideshow--next-arrow-hover:regular"
2015-05-17 10:55:31 - verbose: Added shape "slideshow--next-arrow:regular"
2015-05-17 10:55:31 - info: Compiling 2 shapes ...
2015-05-17 10:55:31 - info: Laying out «css» sprite («css» mode)
2015-05-17 10:55:31 - verbose: Created «css» SVG sprite file («css» mode)
2015-05-17 10:55:31 - verbose: Created «styl» stylesheet resource
2015-05-17 10:55:31 - info: Finished «css» sprite compilation
2015-05-17 10:55:31 - info: Created 1 x SVG, 1 x STYL
2015-05-17 10:55:31 - verbose: Created sprites-4c00f527.svg: 637 Bytes
2015-05-17 10:55:31 - verbose: Created sprites-svg.styl: 363 Bytes
```

**Comment by jkphl** [comment]
Thanks for reporting this. Could you please post your spriter config as well? Thanks & cheers, Joschi

**Comment by donut** [comment]
I think I just updated my question with what you need. Let me know if that's not it.

**Comment by jkphl** [comment]
Ok, great. I'm only on my mobile right now, but will look into it as soon as possible. Btw. did you  already try to set logging to verbose mode?

**Comment by donut** [comment]
Updated my question with the output from verbose logging.

**Comment by donut** [comment]
Further experimenting showed that I can rotate the "missing" next-arrow.svg file and it shows up rotated, just with the wrong color. If next-arrow.svg is the only SVG in the sprite sheet it shows up the correct color.

**Comment by donut** [comment]
I think I may have found a solution. Just yesterday I upgraded to the 2015 version of Adobe Illustrator. This overwrote some of my preferences including the settings I use when saving SVG files. Specifically, I found that in the SVG Options dialog that comes up when saving an SVG, under "Advanced Options" setting "CSS Properties" to "Presentation Attributes" solves the problem. Before I had it set to "Style Elements".

Looking at the generated SVG files, they now have the fill set with an attribute instead of a CSS style definition. I think the problem was that both SVG shapes had the same ID and so the later one's fill settings applied to both shapes.

**Comment by jkphl** [comment]
Any chance your problem relates to this: https://github.com/flobacher/svg-sprite/commit/25f4a45c90f6b384fbe932b1c6cf9279f29a3534#commitcomment-11662593 ?

**Comment by donut** [comment]
That looks like it is exactly the same problem.

**Comment by jkphl** [comment]
Yeah, that's what I thought. So then you'll have two options I guess:
1.) Wait until I added class namespacing. I will do it, but it probably won't be very soon (I'm just lacking the time ...)
2.) Use the tool [extract-svg-styles](https://github.com/ferdinandtorggler/extract-svg-styles) mentioned [in the other issue](https://github.com/flobacher/svg-sprite/commit/25f4a45c90f6b384fbe932b1c6cf9279f29a3534) to extract the styles out of your SVGs.

**Comment by donut** [comment]
So far it looks like exporting the SVGs with the "Presentation Attributes" solution mentioned above is working fine for me. I'll keep this in mind in case this doesn't work outside of Chrome.

**Comment by jkphl** [comment]
Ok then. And keep an eye on the [namespacing issue](https://github.com/jkphl/svg-sprite/issues/42) if you like.
Cheers, Joschi

**Comment by jkphl** [comment]
@donut FYI: CSS class namespacing is now available as of [release 1.2.26](https://github.com/jkphl/svg-sprite/releases/tag/v1.2.6). :)

**Comment by donut** [comment]
Great! Thanks for the update.
